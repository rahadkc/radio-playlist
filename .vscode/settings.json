{
    "files.associations": {
        "**/App.tsx": "typescriptreact",
        "**/App.js": "jsx"
    },
    "typescript.tsdk": "node_modules/typescript/lib"
}