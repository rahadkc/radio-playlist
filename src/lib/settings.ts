enum settings {
  CONTAINER_SIZE = '640px',
  STATION_HEADER_TITLE = 'STATIONS',
  STATION_FOOTER_TITLE = 'Currently Playing',
  LOADING_TEXT = 'Loading...',
  NO_DATA_AVAILABLE_TEXT = ':) No Radio Station available'
}

export default settings
