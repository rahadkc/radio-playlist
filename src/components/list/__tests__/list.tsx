import { cleanup, render, screen } from '@testing-library/react'
import settings from '../../../lib/settings'
import List from '../index'

const mockData: MockType[] = [
  { id: 1, name: 'Messi', value: 200.6 },
  { id: 2, name: 'C Ronaldo', value: '101.5' },
  { id: 3, name: 'Mbappe', value: 200 }
]

type MockType = {
  id: number
  name: string
  value: number | string
}

const text = 'Mock no data'

describe('<List>', () => {
  afterEach(cleanup)

  test('render List component', () => {
    const element = render(<List loading={false} data={mockData} />)
    expect(screen.queryByText(/loading/i)).not.toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })

  test('<Loading> show when loading is true', () => {
    const element = render(<List loading={true} data={mockData} />)
    expect(screen.queryByText(/loading/i)).toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })

  test('<noData> component will not show with atleast one item length', () => {
    const element = render(<List loading={false} data={mockData} noDataText={text} />)
    expect(element.queryByText(text)).not.toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })

  test('<NoData> change component text', () => {
    const element = render(<List loading={false} data={[]} noDataText={text} />)
    expect(element.queryByText(text)).toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })

  test('<noData> show text without noData prop', () => {
    const element = render(<List loading={false} data={[]} />)
    expect(element.queryByText(settings.NO_DATA_AVAILABLE_TEXT)).toBeInTheDocument()
    expect(element).toMatchSnapshot()
  })
})
