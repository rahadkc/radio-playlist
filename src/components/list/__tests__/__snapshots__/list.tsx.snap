// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List> <Loading> show when loading is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loading"
      >
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loading"
    >
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<List> <NoData> change component text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="noData"
      >
        <p
          class="para"
        >
          Mock no data
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="noData"
    >
      <p
        class="para"
      >
        Mock no data
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<List> <noData> component will not show with atleast one item length 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="height: 100%; outline: none; overflow-y: auto; position: relative;"
        tabindex="0"
      >
        <div
          style="width: 100%; height: 100%; position: absolute; top: 0px;"
        >
          <div
            style="box-sizing: border-box; padding-top: 0px; padding-bottom: 0px; margin-top: 0px;"
          >
            <div
              data-index="0"
              data-item-index="0"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="1"
              data-item-index="1"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="2"
              data-item-index="2"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="3"
              data-item-index="3"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="4"
              data-item-index="4"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="5"
              data-item-index="5"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="6"
              data-item-index="6"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="7"
              data-item-index="7"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="8"
              data-item-index="8"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="9"
              data-item-index="9"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="height: 100%; outline: none; overflow-y: auto; position: relative;"
      tabindex="0"
    >
      <div
        style="width: 100%; height: 100%; position: absolute; top: 0px;"
      >
        <div
          style="box-sizing: border-box; padding-top: 0px; padding-bottom: 0px; margin-top: 0px;"
        >
          <div
            data-index="0"
            data-item-index="0"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="1"
            data-item-index="1"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="2"
            data-item-index="2"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="3"
            data-item-index="3"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="4"
            data-item-index="4"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="5"
            data-item-index="5"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="6"
            data-item-index="6"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="7"
            data-item-index="7"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="8"
            data-item-index="8"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="9"
            data-item-index="9"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<List> <noData> show text without noData prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="noData"
      >
        <p
          class="para"
        >
          :) No Radio Station available
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="noData"
    >
      <p
        class="para"
      >
        :) No Radio Station available
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<List> render List component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="height: 100%; outline: none; overflow-y: auto; position: relative;"
        tabindex="0"
      >
        <div
          style="width: 100%; height: 100%; position: absolute; top: 0px;"
        >
          <div
            style="box-sizing: border-box; padding-top: 0px; padding-bottom: 0px; margin-top: 0px;"
          >
            <div
              data-index="0"
              data-item-index="0"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="1"
              data-item-index="1"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="2"
              data-item-index="2"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="3"
              data-item-index="3"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="4"
              data-item-index="4"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="5"
              data-item-index="5"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="6"
              data-item-index="6"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="7"
              data-item-index="7"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="8"
              data-item-index="8"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
            <div
              data-index="9"
              data-item-index="9"
              data-known-size="0"
            >
              <div>
                hello
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="height: 100%; outline: none; overflow-y: auto; position: relative;"
      tabindex="0"
    >
      <div
        style="width: 100%; height: 100%; position: absolute; top: 0px;"
      >
        <div
          style="box-sizing: border-box; padding-top: 0px; padding-bottom: 0px; margin-top: 0px;"
        >
          <div
            data-index="0"
            data-item-index="0"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="1"
            data-item-index="1"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="2"
            data-item-index="2"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="3"
            data-item-index="3"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="4"
            data-item-index="4"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="5"
            data-item-index="5"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="6"
            data-item-index="6"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="7"
            data-item-index="7"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="8"
            data-item-index="8"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
          <div
            data-index="9"
            data-item-index="9"
            data-known-size="0"
          >
            <div>
              hello
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
